src/components/Mentees/EditMentee.js

```diff
@@ -0,0 +1,339 @@
+   * or false when some data is missing.
+   * @returns {Object} the Firebase payload
+   */
+  allRequiredDataProvided = () => {
+    const { descendent, location } = this.state;
+    const fieldsAreRequired = descendent === "" || location === "";
+    if (fieldsAreRequired) {
+      this.setState({
+        error: "All fields are required",
+        openSnackbarError: true
+      });
+      return false;
+    }
+    return true;
+  };
+
+  /**
+   * handleSubmit - review if all required data has been provided and then create
+   * an account with the email and password.
+   * @param {Object} e the event object
+   * @param {Object} reason for closing the snackbar
+   * @return {void}
+   */
+  handleSubmit = e => {
+    e.preventDefault();
+
+    if (this.allRequiredDataProvided()) {
+      editMentee(this.getFirebasePayload(), auth.currentUserUid())
+        .then(() => {
+          this.setState({
+            openSnackbarSaved: true,
+            sectionError: ""
+          });
+        })
+        .catch(error => {
+          this.setState({
+            error: error.message,
+            openSnackbarError: true
+          });
+        });
+    }
+  };
+  /**
+   * areThereParameters – sets the state with the parameters sent via url
+   * @returns {void}
+   *
+   */
+  getMenteeData = () => {
+    getMentee(auth.currentUserUid())
+      .then(mentee => {
+        this.setState({
+          name: mentee.val().name,
+          email: mentee.val().email,
+          descendent: mentee.val().descendent,
+          location: mentee.val().location
+        });
+      })
+      .catch();
+  };
+  /**
+   * componentDidMount – sets in the state data to edit
+   * @returns {void}
+   */
+  componentDidMount = () => {
+    this.unregisterObserver = this.getMenteeData();
+  };
+
+  /**
+   * componentWillUnmount – clean the functions loaded in componentDidMount in
+   * order to avoid leaks of memory.
+   * @returns {void}
+   */
+  componentWillUnmount() {
+    this.unregisterObserver = null;
+  }
+
+  /**
+   * handleSnackbarClose - sets the actions when the snackbar is closed
+   * @param {Object} event the event object
+   * @param {Object} reason for closing the snackbar
+   * @return {void}
+   */
+  handleSnackbarClose = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+    this.state.openSnackbarSaved
+      ? this.setState({
+          openSnackbarSaved: false,
+          returnMentor: true
+        })
+      : this.setState({ openSnackbarError: false });
+  };
+
+  render() {
+    const { classes } = this.props;
+    const {
+      name,
+      location,
+      error,
+      descendent,
+      openSnackbarError,
+      nameError,
+      descendentError,
+      locationError,
+      openSnackbarSaved
+    } = this.state;
+
+    return (
+      <div className={classes.wrapper}>
+        <Card className={classes.card}>
+          <form onSubmit={this.handleSubmit}>
+            <CardContent>
+              <div>
```

Comment:
This is just a suggestion. We can remove the card component here since the panel creates a background, we don't need to have a card inside another card. Same in the EditMentor component.

Reply:
Good idea, it looks better that way. Done!